tensorflow_probability/python/distributions/pert_test.py

```diff
@@ -0,0 +1,205 @@
+              .sample(n, seed=tfp_test_util.test_seed())
+    samples = self.evaluate(dist)
+    expected_mean = sp_stats.beta.mean(a, b, mini, maxi-mini)
+    expected_var = sp_stats.beta.var(a, b, mini, maxi-mini)
+    self.assertAllClose(
+        samples.mean(axis=0),
+        expected_mean,
+        rtol=0.01,
+        atol=0.01,
+        msg="Sample mean is highly off of expected.")
+    self.assertAllClose(
+        samples.var(axis=0),
+        expected_var,
+        rtol=0.01,
+        atol=0.01,
+        msg="Sample variance is highly off of expected.")
+
+  # Parameter restriction testing
+  def testSmoothnessPositive(self):
+    smoothness = tf.Variable(0.)
+    mini = [1., 2., 3.]
+    mode = [2., 3., 4.]
+    maxi = [3., 4., 5.]
+    self.evaluate(smoothness.initializer)
+    with self.assertRaisesRegex(
+        tf.errors.InvalidArgumentError,
+        "Smoothness parameter must be positive."):
+      dist = tfd.PERT(mini, mode, maxi, smoothness, validate_args=True)
+      self.evaluate(dist.sample(1))
+
+  def testSmoothnessPositiveAfterMutation(self):
+    smoothness = tf.Variable(4.)
+    mini = [1., 2., 3.]
+    mode = [2., 3., 4.]
+    maxi = [3., 4., 5.]
+    self.evaluate(smoothness.initializer)
+    dist = tfd.PERT(mini, mode, maxi, smoothness, validate_args=True)
+    with self.assertRaisesRegex(
+        tf.errors.InvalidArgumentError,
+        "Smoothness parameter must be positive."):
+      with tf.control_dependencies([smoothness.assign(-1.)]):
+        self.evaluate(dist.sample(1))
+
+  def testModeMinInequality(self):
+    mini = tf.Variable([1., 2., 3.])
+    mode = tf.Variable([1., 2., 3.])
+    maxi = [5., 5., 5.]
+    self.evaluate(mini.initializer)
+    self.evaluate(mode.initializer)
+    with self.assertRaisesRegex(
+        tf.errors.InvalidArgumentError,
+        "Mode must be greater than minimum."):
+      dist = tfd.PERT(mini, mode, maxi, validate_args=True)
+      self.evaluate(dist.sample(1))
+
+  def testModeMinInequalityAfterMutation(self):
+    mini = tf.Variable([1., 2., 3.])
+    mode = tf.Variable([2., 3., 4.])
+    maxi = [5., 5., 5.]
+    self.evaluate(mini.initializer)
+    self.evaluate(mode.initializer)
+    dist = tfd.PERT(mini, mode, maxi, validate_args=True)
+    with self.assertRaisesRegex(
+        tf.errors.InvalidArgumentError,
+        "Mode must be greater than minimum."):
+      with tf.control_dependencies([mode.assign([0., 0., 0.])]):
+        self.evaluate(dist.sample(1))
+
+  def testMaxModeInequality(self):
+    mini = [1., 2., 3.]
+    mode = tf.Variable([2., 3., 4.])
+    maxi = tf.Variable([5., 5., 4.])
+    self.evaluate(maxi.initializer)
+    self.evaluate(mode.initializer)
+    with self.assertRaisesRegex(
+        tf.errors.InvalidArgumentError,
+        "Maximum must be greater than mode."):
+      dist = tfd.PERT(mini, mode, maxi, validate_args=True)
+      self.evaluate(dist.sample(1))
+
+  def testMaxModeInequalityAfterMutation(self):
```

Comment:
this test looks great!

if you want a little more coverage, you can add a 
```python
@classmethod
def _params_event_ndims(cls):
  return dict(low=0, high=0, peak=0, temperature=0)
```
to the distribution (which will also make it batch-sliceable, i.e. `tfd.PERT(...)[..., 3:, -2]`), and then whitelist it in distribution_properties_test.py

Reply:
Cool, I haven't fully wrapped my head around details of `distribution_properties_test.py` but I get the gist of it.

https://github.com/tensorflow/probability/blob/ba0701f00cd1bed083caa61f392f4acd91f0fe20/tensorflow_probability/python/distributions/distribution_properties_test.py#L1048

I saw the test inputting some wild parameters, so I fixed it by adding something similar to `fix_triangular` with additional handling of `temperature`.

Reply:
Great, I'll double check it when I bring the code in. I wonder if rebasing the PR instead of adding the commits into this PR would fix the CLA issue. It seems to have a problem with the "no-reply@google.com" commit (I think, the PlackettLuce one).